apps/wallet/src/ui/app/components/transactions-card/TxnImage.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Text } from '_app/shared/text';
+import { useGetNFTMeta, useMiddleEllipsis } from '_hooks';
+
+const TRUNCATE_MAX_CHAR = 34;
+
+//TODO merge all NFT image displays
+export function TxnImage({ id }: { id: string }) {
+    const nftMeta = useGetNFTMeta(id);
+    const name = useMiddleEllipsis(
```

Comment:
I feel like truncation should happen on the end not the middle right? Makes more sense to me to lean into CSS for this instead of using middle ellipsis truncation.

Reply:
done